modules/packages/LinearAlgebra.chpl

```diff
@@ -1153,17 +1210,45 @@ proc eigvals(A: [] ?t, param left = false, param right = false)
 /*
   Singular Value Decomposition.
 
-  Factorizes the matrix ``A`` into two unitary matrices, ``U`` and ``Vt`` and
-  a vector of singular values ``s``, such that::
+  Factorizes the `m x n` matrix ``A`` such that:
+
+
+  .. math::
+
+    \mathbf{A} = \textbf{U} \cdot \Sigma \cdot \mathbf{V^H}
 
-    A = U * s * Vt
+  where
 
-  Will throw a ``LinearAlgebraError`` if the SVD computation does not converge
-  or an illegal argument (such as ``NAN``) is given.
+    - :math:`\mathbf{U}` is an `m x m` unitary matrix,
+    - :math:`\Sigma` is a diagonal `m x n` matrix,
+    - :math:`\mathbf{V}` is an `n x n` unitary matrix, and
+      :math:`\mathbf{V^H}` is the Hermitian transpose.
+
+  This procedure returns a tuple of ``(U, s, Vh)``, where ``s`` is a vector
+  containing the diagonal elements of :math:`\Sigma`, known as the
+  singular values.
+
+  For example:
+
+  .. code-block:: chapel
+
+    var A = Matrix([3, 2,  2],
+                   [2, 3, -2],
+                   eltType=real);
```

Comment:
Can you use:

```chpl
var A = Matrix([3.0, 2.0,  2.0],
               [2.0, 3.0, -2.0])
```

instead of `eltType=real`?

Reply:
Yes, although I was implicitly using this as an opportunity to show what users can do with the `Matrix()` interface. Maybe showing one of the examples without the `eltType` argument would be good as well.

Reply:
Ah, ok. Up to you